libs/indibase/indiccdchip.cpp

```diff
@@ -117,38 +117,50 @@ void CCDChip::setFrame(uint32_t subx, uint32_t suby, uint32_t subw, uint32_t sub
     SubW = subw;
     SubH = subh;
 
-    ImageFrameN[FRAME_X].value = SubX;
-    ImageFrameN[FRAME_Y].value = SubY;
-    ImageFrameN[FRAME_W].value = SubW;
-    ImageFrameN[FRAME_H].value = SubH;
+    ImageFrameNP[FRAME_X].setValue(SubX);
+    ImageFrameNP[FRAME_Y].setValue(SubY);
+    ImageFrameNP[FRAME_W].setValue(SubW);
+    ImageFrameNP[FRAME_H].setValue(SubH);
 
-    IDSetNumber(&ImageFrameNP, nullptr);
+    ImageFrameNP.apply();
 }
 
 void CCDChip::setBin(uint8_t hor, uint8_t ver)
 {
     BinX = hor;
     BinY = ver;
 
-    ImageBinN[BIN_W].value = BinX;
-    ImageBinN[BIN_H].value = BinY;
+    ImageBinNP[BIN_W].setValue(BinX);
+    ImageBinNP[BIN_H].setValue(BinY);
 
-    IDSetNumber(&ImageBinNP, nullptr);
+    ImageBinNP.apply();
 }
 
 void CCDChip::setMinMaxStep(const char *property, const char *element, double min, double max, double step,
                             bool sendToClient)
 {
     INumberVectorProperty *nvp = nullptr;
 
-    if (!strcmp(property, ImageExposureNP.name))
-        nvp = &ImageExposureNP;
-    else if (!strcmp(property, ImageFrameNP.name))
-        nvp = &ImageFrameNP;
-    else if (!strcmp(property, ImageBinNP.name))
-        nvp = &ImageBinNP;
-    else if (!strcmp(property, ImagePixelSizeNP.name))
-        nvp = &ImagePixelSizeNP;
+    auto updateMinMaxStep = [element, min, max, sendToClient](INDI::PropertyNumber &oneProperty)
+    {
+        auto oneElement = oneProperty.findWidgetByName(element);
+        if(oneElement)
+        {
+            oneElement->setMinMax(min, max);
+            if(sendToClient)
+                oneProperty.updateMinMax();
+        }
+    };
+
+    if (ImageExposureNP.isNameMatch(property))
+        updateMinMaxStep(ImageExposureNP);
+    else if (ImageFrameNP.isNameMatch(property))
+        updateMinMaxStep(ImageFrameNP);
+    else if (ImageBinNP.isNameMatch(property))
+        updateMinMaxStep(ImageBinNP);
+
+    else if (ImagePixelSizeNP.isNameMatch(property))
+        updateMinMaxStep(ImagePixelSizeNP);
     //    else if (!strcmp(property, RapidGuideDataNP.name))
     //        nvp = &RapidGuideDataNP;
     else
```

Comment:
remove the last else, it should be just return

Also remove the code below since it's never executed